tests/unit/states/test_vmc_security_groups.py

```diff
@@ -0,0 +1,724 @@
+                        "key": "OSName",
+                        "operator": "EQUALS",
+                        "value": "Centos",
+                        "resource_type": "Condition",
+                    }
+                ],
+                "tags": [{"tag": "tag1", "scope": "scope1"}],
+            }
+        ),
+    ],
+)
+def test_present_state_during_create_should_correctly_pass_args(mocked_ok_response, actual_args):
+    mocked_updated_response = mocked_ok_response.copy()
+    mock_get_by_id_response = create_autospec(vmc_security_groups_exec.get_by_id, return_value={})
+    mock_create_response = create_autospec(
+        vmc_security_groups_exec.create, return_value=mocked_ok_response
+    )
+
+    common_actual_args = {
+        "hostname": "hostname",
+        "refresh_key": "refresh_key",
+        "authorization_host": "authorization_host",
+        "org_id": "org_id",
+        "sddc_id": "sddc_id",
+        "domain_id": "domain_id",
+        "security_group_id": mocked_ok_response["id"],
+        "verify_ssl": False,
+    }
+
+    mocked_updated_response.update(actual_args)
+    actual_args.update(common_actual_args)
+    mock_create = create_autospec(
+        vmc_security_groups_exec.create, return_value=mocked_updated_response
+    )
+
+    with patch.dict(
+        vmc_security_groups.__salt__,
+        {
+            "vmc_security_groups.get_by_id": mock_get_by_id_response,
+            "vmc_security_groups.create": mock_create,
+        },
+    ):
+        result = vmc_security_groups.present(name="secuirty group create", **actual_args)
+
+    call_kwargs = mock_create.mock_calls[0][-1]
+
+    subset = {k: v for k, v in call_kwargs.items() if k in actual_args}
+    assert subset == actual_args
+
+    assert result is not None
+    assert result["changes"]["old"] is None
+    assert result["changes"]["new"] == mocked_updated_response
+    assert result["comment"] == "Created Security group security_group_id"
+    assert result["result"]
+
+
+def test_present_state_for_update_when_opts_test_is_true(mocked_ok_response):
+    mock_get_by_id_response = create_autospec(
+        vmc_security_groups_exec.get_by_id, return_value=mocked_ok_response
+    )
+    security_group_id = mocked_ok_response["id"]
+
+    with patch.dict(
+        vmc_security_groups.__salt__,
+        {"vmc_security_groups.get_by_id": mock_get_by_id_response},
+    ):
+        with patch.dict(vmc_security_groups.__opts__, {"test": True}):
+            result = vmc_security_groups.present(
+                name="test_present",
+                hostname="nsx-t.vmwarevmc.com",
+                refresh_key="refresh_key",
+                authorization_host="authorization_host",
+                org_id="org_id",
+                sddc_id="sddc_id",
+                domain_id="domain_id",
+                security_group_id=security_group_id,
+            )
+
+    assert result is not None
+    assert len(result["changes"]) == 0
```

Comment:
**suggestion (non-blocking)** This could be written as `assert not result['changes']`, which might read better.

Reply:
updated